theano/tensor/basic.py

```diff
@@ -1715,6 +1715,11 @@ def min(x, axis=None, keepdims=False):
     str_x_type = str(x.dtype)
     if str_x_type.startswith('float') or str_x_type in int_dtypes:
         return -max(-x, axis=axis, keepdims=keepdims)
+    elif str_x_type in uint_dtypes:
+        itype = np.iinfo(x.dtype)
+        return itype.max - max(itype.max - x, axis=axis, keepdims=keepdims)
```

Comment:
The only problem I have with that is that it upcast the result to an higher dtype then the original dtype.

I think this can be easily fixed by this change:

    itype.max -> np.asarray(itype.max, dtype=x.dtype)

Can you try that and add in the test an assert that the output dtype is the same as the input dtype?

Reply:
Good point, fixed.